codec-http/src/main/java/io/netty/handler/codec/spdy/SpdyFrameDecoder.java

```diff
@@ -421,6 +431,37 @@ private static State getNextState(int type, int length) {
         }
     }
 
+    /**
+     * Decode the unknown frame, returns true if parsed something, otherwise false.
+     */
+    protected boolean decodeUnknownFrame(int frameType, byte flags, int length, ByteBuf buffer) {
+        if (length == 0) {
+            if (delegate instanceof SpdyFrameDecoderExtendedDelegate) {
+                ((SpdyFrameDecoderExtendedDelegate) delegate).readUnknownFrame(frameType, flags, Unpooled.EMPTY_BUFFER);
+            }
+            return true;
+        }
+        if (buffer.readableBytes() < length) {
+            return false;
+        }
+        if (delegate instanceof SpdyFrameDecoderExtendedDelegate) {
+            ByteBuf data = buffer.alloc().buffer(length);
```

Comment:
what about using `buffer.readSlice` here?

Reply:
Yes we could use `readRetainedSlice(...)`. But let's do this in a followup as its true for other methods as well.